src/main/kotlin/org/partiql/scribe/problems/ScribeProblem.kt

```diff
@@ -0,0 +1,93 @@
+package org.partiql.scribe.problems
+
+import org.partiql.spi.Enum
+import org.partiql.spi.UnsupportedCodeException
+
+/**
+ * Represents a problem that is encountered during transpilation.
+ *
+ * @property code
+ * @property severity The severity of the problem.
+ * @property properties The properties of the problem.
+ */
+public class ScribeProblem(
+    private val code: Int,
+    public val severity: Severity,
+    private val properties: Map<String, Any>,
+) : Enum(code) {
+    override fun name(): String {
+        return when (code) {
+            INTERNAL_ERROR -> "INTERNAL_ERROR"
+            UNSUPPORTED_OPERATION -> "UNSUPPORTED_OPERATION"
+            UNSUPPORTED_PLAN_TO_AST_CONVERSION -> "UNSUPPORTED_PLAN_TO_AST_CONVERSION"
+            INVALID_PLAN -> "INVALID_PLAN"
```

Comment:
self-review: for now, these are the `code`s we support. we can change the codes along with supported properties as we continue work on Scribe